### PR TITLE
Fixes #37431 - Fix Style/SoleNestedConditional cop

### PR DESCRIPTION
**Pull request by archanaserver**
Reviewed and optimized nested conditionals flagged by the Style/SoloNestedConditional cop.

### CODE REVIEW

**Review by nofaralfasi — CHANGES_REQUESTED** [review]
The failing tests appear to be related to your changes. Additionally, after applying these changes, there is a need to fix the indentations.

File: app/controllers/concerns/foreman/controller/users_mixin.rb [hunk 1]
```diff
@@ -13,9 +13,9 @@ def resource_scope(options = {})
   protected
 
   def clear_session_locale_on_update
-    if params[:user] && editing_self?
+    if params[:user] && editing_self? && params[:user][:locale].try(:empty?)
```
Comment (nofaralfasi) [inline app/controllers/concerns/foreman/controller/users_mixin.rb]:
```suggestion
    user_params = params[:user]
    if user_params && editing_self? && user_params[:locale].blank?
```

File: app/models/compute_resources/foreman/model/ovirt.rb [hunk 2]
```diff
@@ -619,11 +619,9 @@ def create_interfaces(vm, attrs, cluster_id)
         raise Foreman::Exception.new("Interface network or vnic profile are missing.") if (interface[:network].nil? && interface[:vnic_profile].nil?)
         interface[:network] = get_ovirt_id(cluster_networks, 'network', interface[:network]) if interface[:network].present?
         interface[:vnic_profile] = get_ovirt_id(profiles, 'vnic profile', interface[:vnic_profile]) if interface[:vnic_profile].present?
-        if (interface[:network].present? && interface[:vnic_profile].present?)
-          unless (profiles.select { |profile| profile.network.id == interface[:network] }).present?
+        if (interface[:network].present? && interface[:vnic_profile].present?) && !(profiles.select { |profile| profile.network.id == interface[:network] }).present?
```
Comment (nofaralfasi) [inline app/models/compute_resources/foreman/model/ovirt.rb]:
```suggestion
        if interface[:network].present? && interface[:vnic_profile].present? && profiles.none? { |profile| profile.network.id == interface[:network] }

```

File: app/models/concerns/foreman/sti.rb [hunk 3]
```diff
@@ -9,12 +9,10 @@ class << base
     module ClassMethods
       # ensures that the correct STI object is created when :type is passed.
       def new(attributes = nil, &block)
-        if attributes.is_a?(Hash) && (type = attributes.with_indifferent_access.delete(:type)) && !type.empty?
-          if (klass = type.constantize) != self
+        if attributes.is_a?(Hash) && (type = attributes.with_indifferent_access.delete(:type)) && !type.empty? && ((klass = type.constantize) != self)
```
Comment (nofaralfasi) [inline app/models/concerns/foreman/sti.rb]:
```suggestion
        if attributes.is_a?(Hash) && (type = attributes.with_indifferent_access.delete(:type)).present? && (klass = type.constantize) != self
```

File: app/models/concerns/orchestration.rb [hunk 4]
```diff
@@ -170,8 +170,8 @@ def process(queue_name)
 
     rollback
   ensure
-    unless q.nil?
-      if processed > 0
+    # rubocop:enable Rails/FindEach
+if !q.nil? && (processed > 0)
```
Comment (nofaralfasi) [inline app/models/concerns/orchestration.rb]:
```suggestion
    if q && processed.positive?
```

File: app/registries/menu/item.rb [hunk 5]
```diff
@@ -23,8 +23,8 @@ def initialize(name, options)
     end
 
     def to_hash
-      if @condition.present?
-        return unless @condition.call
+      if @condition.present? && !@condition.call
+        return
       end
```
Comment (nofaralfasi) [inline app/registries/menu/item.rb]:
```suggestion
  return if @condition.present? && !@condition.call
```

File: app/services/ui_notifications/rss_notifications_checker.rb [hunk 6]
```diff
@@ -47,8 +47,8 @@ def deliver!
       feed.items[0, @latest_posts].each do |feed_item|
         item = Item.new(feed_item)
         blueprint = rss_notification_blueprint
-        if notification_already_exists?(item)
-          next unless @force_repost
+        if notification_already_exists?(item) && !@force_repost
+          next
         end
```
Comment (nofaralfasi) [inline app/services/ui_notifications/rss_notifications_checker.rb]:
```suggestion
        next if notification_already_exists?(item) && !@force_repost
```

File: lib/net/dhcp/record.rb [hunk 7]
```diff
@@ -74,8 +74,8 @@ def eql_for_conflicts?(other)
       # If we're converting an 'ad-hoc' lease created by a host booting outside of Foreman's knowledge,
       # then :hostname will be blank on the incoming lease - if the ip/mac still match, then this
       # isn't a conflict. Only applicable on legacy proxy API without "type" attribute.
-      if legacy_dhcp_api?
-        to_compare << :hostname if other.attrs[:hostname].present? && attrs[:hostname].present?
+      if legacy_dhcp_api? && (other.attrs[:hostname].present? && attrs[:hostname].present?)
+        to_compare << :hostname
       end
```
Comment (nofaralfasi) [inline lib/net/dhcp/record.rb]:
```suggestion
      to_compare << :hostname if legacy_dhcp_api? && other.attrs[:hostname].present? && attrs[:hostname].present?
```

File: app/models/concerns/orchestration.rb [hunk 8]
```diff
@@ -170,20 +170,19 @@ def process(queue_name)
...
-          if task.status?(:completed) || task.status?(:pending)
-            logger.debug msg
-          else
-            logger.error msg
-          end
+    # rubocop:enable Rails/FindEach
+    if !q.nil? && (processed > 0)
+      logger.info("Processed #{processed} tasks from queue '#{q.name}', completed #{q.completed.count}/#{q.all.count}") unless q.empty?
+      # rubocop:disable Rails/FindEach
+      q.all.each do |task|
```
Comment (ekohl) [inline app/models/concerns/orchestration.rb]: I think you can also disable it with just this. Then you can avoid the other control statements.
```suggestion
      q.all.each do |task|# rubocop:disable Rails/FindEach
```

File: app/controllers/api/base_controller.rb [hunk 9]
```diff
@@ -308,7 +308,6 @@ def setup_search_options
             query = " #{Regexp.last_match(1)} = #{value}"
             params[:search] += query unless params[:search].include? query
           end
-        end
```
Comment (ekohl) [inline app/controllers/api/base_controller.rb:311]: This looks like it's missing some other change

**Comment by archanaserver** [comment]
@ekohl there is one rubocop offense detected that is `Lint/Syntax` https://github.com/theforeman/foreman/actions/runs/9093977252/job/24994075137?pr=10068, even though syntax seems correct to me when i checked the file. Do you mind checking the failure here?

File: app/controllers/concerns/foreman/controller/users_mixin.rb [hunk 10]
```diff
@@ -13,7 +13,8 @@ def resource_scope(...)
   protected
 
   def clear_session_locale_on_update
-    if params[:user] && editing_self?
+    user_params = params[:user]
+    if user_params && editing_self? && user_params[:locale].blank?
       # Remove locale from the session when set to "Browser Locale" and editing self
       session.delete(:locale) if params[:user][:locale].try(:empty?)
```
Comment (ekohl) [inline app/controllers/concerns/foreman/controller/users_mixin.rb]: If you add `user_params[:locale].blank?` then you should drop `if params[:user][:locale].try(:empty?)`

File: app/models/concerns/orchestration/dhcp.rb [hunk 11]
```diff
@@ -173,11 +173,9 @@ def dhcp_update_required?
       (!old.build? && build? && !all_dhcp_records_valid?))
     # Handle jumpstart
     # TODO, abstract this way once interfaces are fully used
```
Comment (ekohl) [inline app/models/concerns/orchestration/dhcp.rb:175]: I wonder if this TODO isn't actually solved with 4cbf879ed5957978afd74b3707c8a6b16fa18225. Probably not for this PR, but something to investigate.

**Review by ekohl — CHANGES_REQUESTED** [review]

File: app/models/concerns/orchestration/dhcp.rb (hunk 11 above)
Comment (archanaserver) [reply app/models/concerns/orchestration/dhcp.rb:175]: I tried doing something here https://github.com/theforeman/foreman/pull/10261

File: app/controllers/concerns/foreman/controller/users_mixin.rb (hunk 10 above)
Comment (archanaserver) [reply app/controllers/concerns/foreman/controller/users_mixin.rb]: @ekohl thanks, this make sense. i've updated it.

**Review by ekohl — DISMISSED** [review]

**Comment by ekohl** [comment]
@nofaralfasi have your concerns been addressed?

**Review by nofaralfasi — DISMISSED** [review]

**Comment by archanaserver** [comment]
@ekohl, can we get this in?

**Review by ehelms — APPROVED** [review]

**Pull request merged**